### PR TITLE
[TIR][Schedule] Analysis functions to check if compute_inline and com…

### PR DESCRIPTION
**Pull request by vinx13**
…pute_inline is allowed

Co-authored-by: Siyuan Feng <Hzfengsy@sjtu.edu.cn>
Co-authored-by: Bohan Hou <32121147+spectrometerHBH@users.noreply.github.com>
Co-authored-by: Hongyi Jin <3231950289@qq.com>
Co-authored-by: Ruihang Lai <lairuihangdongdong@qq.com>
Co-authored-by: Junru Shao <junrushao1994@gmail.com>
Co-authored-by: Wuwei Lin <wuwei@apache.org>
Co-authored-by: Xiyou Zhou <xiyou@octoml.ai>

### CODE REVIEW

**Review by junrushao1994 — APPROVED** [review]
LGTM

File: src/tir/schedule/primitive/compute_at.cc [hunk 1]
```diff
@@ -501,29 +503,59 @@ void ComputeAtOrReverseComputeAtImpl(ScheduleState self, const StmtSRef& block_s
...
   Block new_scope_root = Downcast<Block>(reconstructor(scope_root));
-  // Step 7. Do the actual replacement
-  self->Replace(scope_root_sref, new_scope_root, {{scope_root, new_scope_root}});
-  // Step 8. Update the cached flags
-  BlockInfo& block_info = self->block_info[block_sref];
-  block_info.affine_binding = IsAffineBinding(
-      /*realize=*/reconstructor.new_block_realize_,
-      /*loop_var_ranges=*/LoopDomainOfSRefTreePath(GetRef<StmtSRef>(block_sref->parent)),
-      /*analyzer=*/&analyzer);
+  return [=]() -> void {
```
Comment (comaniac) [inline src/tir/schedule/primitive/compute_at.cc:511]: This style is interesting but I'm not sure if it's intuitive. Would that be better to just have an optional argument like `check_only=false`?

File: src/tir/schedule/primitive/compute_inline.cc [hunk 2]
```diff
@@ -550,17 +567,32 @@ void ComputeInline(ScheduleState self, const StmtSRef& producer_block_sref) {
     throw OpaqueAccessError(self->mod, scope_root_sref);
   }
   // Step 6. Do the real mutation on the AST and the sref tree in the schedule state
-  self->Replace(scope_root_sref, tgt_stmt, inliner.block_reuse);
+  return [=]() -> void { self->Replace(scope_root_sref, tgt_stmt, inliner.block_reuse); };
```
Comment (comaniac) [inline src/tir/schedule/primitive/compute_inline.cc:570]: ditto

File: src/tir/schedule/primitive/compute_inline.cc [hunk 3]
```diff
@@ -579,7 +611,20 @@ void ReverseComputeInline(ScheduleState self, const StmtSRef& consumer_block_sre
     throw OpaqueAccessError(self->mod, scope_root_sref);
   }
   // Step 7. Do the real mutation on the AST and the sref tree in the schedule state
-  self->Replace(scope_root_sref, tgt_stmt, inliner.block_reuse);
+  return [=]() -> void { self->Replace(scope_root_sref, tgt_stmt, inliner.block_reuse); };
```
Comment (comaniac) [inline src/tir/schedule/primitive/compute_inline.cc:614]: ditto

**Review by comaniac — COMMENTED** [review]
Overall LGTM. Just nits.

File: src/tir/schedule/primitive/compute_at.cc (hunk 1 above)
Comment (junrushao1994) [reply src/tir/schedule/primitive/compute_at.cc]: Why do you both hate my fashionable coolest functional style 😭😝

**Review by comaniac — APPROVED** [review]
LGTM

**Review by Hzfengsy — APPROVED** [review]
LGTM

**Pull request merged**